### PR TITLE
JAMES-2586 Implement PostgresMailRepository

### PR DESCRIPTION
**Pull request by quantranhong1999**
resolve https://github.com/linagora/james-project/issues/4980

continuation of: https://github.com/apache/james-project/pull/1853

TODO:
- [x] Fix PER_RECIPIENT_SPECIFIC_HEADERS column reading (jooq compilation and runtime return type is not the same @@)
- [x] Guice registrations for postgres protocol for mail repository using PostgresMailRepository
- [x] Adding all field support for Mail
- [x] Removing JPAMailRepository & related classes
- [x] Updating postgres-app default configuration

### CODE REVIEW

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/ForwardIntegrationTest.java [hunk 1]
```diff
@@ -19,10 +19,11 @@
 
 package org.apache.james.mailets;
 
-import static org.apache.james.MemoryJamesServerMain.SMTP_ONLY_MODULE;
+import static org.apache.james.MemoryJamesServerMain.SMTP_AND_IMAP_MODULE;
```
Comment (chibenwa) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/ForwardIntegrationTest.java:22]: UNRELATED changes

**Comment by quantranhong1999** [comment]
Well, our storage fully relies on Postgres now (not anymore JPA part).

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]
